nautobot_ssot/contrib.py

```diff
@@ -0,0 +1,354 @@
+        # This is in fact callable, because it is a model
+        obj = cls._model()  # pylint: disable=not-callable
+
+        cls._update_obj_with_parameters(obj, parameters)
+
+        return super().create(diffsync, ids, attrs)
+
+    @classmethod
+    def _update_obj_with_parameters(cls, obj, parameters):
+        """Update a given Nautobot ORM object with the given parameters."""
+        # Example: {"group": {"name": "Group Name", "_model_class": TenantGroup}}
+        foreign_keys = defaultdict(dict)
+
+        # Example: {"tags": [Tag-1, Tag-2]}
+        many_to_many_fields = defaultdict(list)
+
+        # Use type hints at runtime to determine which fields are custom fields
+        type_hints = get_type_hints(cls, include_extras=True)
+
+        for field, value in parameters.items():
+            cls._check_field(field)
+
+            # Prepare handling of foreign keys
+            if "__" in field:
+                related_model, lookup = field.split("__", maxsplit=1)
+                django_field = cls._model._meta.get_field(related_model)
+                foreign_keys[related_model][lookup] = value
+                # Add a special key to the dictionary to point to the related model's class
+                foreign_keys[related_model]["_model_class"] = django_field.related_model
+                continue
+
+            # Handle custom fields. See CustomFieldAnnotation docstring for more details.
+            is_custom_field = False
+            metadata_for_this_field = getattr(type_hints[field], "__metadata__", [])
+            for metadata in metadata_for_this_field:
+                if isinstance(metadata, CustomFieldAnnotation):
+                    obj.cf[metadata.name] = value
+                    is_custom_field = True
+                    continue
+            if is_custom_field:
+                continue
+
+            django_field = cls._model._meta.get_field(field)
+
+            # Prepare handling of many-to-many fields. If we are dealing with a many-to-many field,
+            # we get all the related objects here to later set them once the object has been saved.
+            if django_field.many_to_many or django_field.one_to_many:
+                many_to_many_fields[field] = [
+                    django_field.related_model.objects.get(**parameters) for parameters in value
+                ]
+                continue
+
+            # As the default case, just set the attribute directly
+            setattr(obj, field, value)
+
+        # Set foreign keys
+        cls._handle_foreign_keys(foreign_keys, obj)
+
+        # Save the object to the database
+        try:
+            obj.validated_save()
+        except ValidationError as error:
+            raise ValidationError(f"Parameters: {parameters}") from error
+
+        # Set many-to-many fields after saving
+        cls._handle_many_to_many_fields(many_to_many_fields, obj)
+
+    @classmethod
+    def _handle_many_to_many_fields(cls, many_to_many_fields, obj):
```

Comment:
similiar to my other comment, here, instead of "handle" may be "set" would be more explicit?